### ISSUE TITLE
Add NFS support to CrateKube

### ISSUE THREAD

**Issue by pdgrenon** [cratekube/cratekube#7]
### Problem to solve
It is difficult to use network attached storage with containers. Within Cisco, NFS via AutoFS has become a standard for legacy storage that is widely used across IT. For many teams, support for NS via AutoFS is absolutely required before they can run their application in containers. 

### Proposal
We should support NFS storage via AutoFS as a capability for CrateKube. To do so, we will need to find a way to dynamically mount the network attached storage into containers at runtime.

Ideally, this feature should be able to:
- Declare NFS storage in Kubernetes YAML
- Mount the NFS storage to a desired location within the container
- Take a username and password to use with authentication to remote NFS storage

The connection to the storage should reconnect after container restart, and migrate with the container if the container moves to a different host.

### Intended users
Anyone who wants to use NFS storage within their containers will find this feature useful. 

### Further details
None.

### Permissions and Security
Whatever mounting strategy we use for the NFS storage will need to be able to support username and password authentication. 

### Documentation
This feature will need user-facing documentation to describe how this feature is used. 

### How can we measure success?
The implementation of this feature is successful if Cisco-internal resources can mount NFS storage into their containers at runtime. 

### When is this feature complete?
This feature is complete when the following items have been completed:

- A new repository has been created to store this feature
- Architecture documentation exists for this feature
- A code-level design exists for this feature
- Automated tests exist for this feature
- This feature has been implemented
- User-facing documentation exists for this feature

### Links / references
There is an existing implementation of this feature that can be used with the Rancher orchestrator for Cisco on-premise environments, but the implementation heavily relies on Cisco-internal infrastructure and cannot be easily ported to CrateKube. This previous implementation is only available within the Cisco network, but may be a useful resource for some maintainers: https://wwwin-github.cisco.com/PDSDockerSquad/automount-service

**Comment by kod4n** [comment]
I believe that we are going to run into firewall restrictions as this wasn't possible for us to do on the previous DMZ installation of Crate.

**Comment by pdgrenon** [comment]
Yeah, there will definitely be some firewall issues to work around to be able to get to Cisco's internal filers, but Cisco Infosec recently came up with a solution to allow this kind of connection. I think we can implement this in a generic way that is still useful to the community and meets the Cisco use case if we leverage that connector.

**Comment by code4dc** [comment]
Concerns:
1. Users need to have AutoFS installed on their hosts to use automounts (for Cisco automounts)
2. IT has a tool for accessing Automounts that we don't know much about (contact Luke) 
3. Allowing external users to access Cisco autmounts is a HUGE security concern

Steps forward:
1. Contact Luke regarding external automount access (Phil) 
2. Set up Kubernetes cluster using RKE on Cisco vSphere hosts (Hogan to grant vSphere access) 
3. Research NFS volumes and whether they allow user/pass
4. POC for NFS automounts on Cisco-internal K8s cluster

**Comment by pdgrenon** [comment]
I will follow up with our security architect regarding our options for the external access requirement.

**Comment by pdgrenon** [comment]
I was able to find the documentation for the connector: https://runon.cisco.com/c/r/runon/Services/SCI/ov/sci_overview.html

*Note* This documentation will only be accessible to users within the Cisco network, as the documentation is for a Cisco-internal service.
